circleguard/gui.py

```diff
@@ -306,15 +310,18 @@ def scroll_to_bottom(self):
     def run(self):
         current_tab = self.tabs.currentIndex()
         self.run_type = MainTab.TAB_REGISTER[current_tab]["name"]
-        thread = threading.Thread(target=self.run_circleguard)
-        thread.start()
+        if not MainTab.TAB_REGISTER[self.tabs.currentIndex()]["custom_action"]:
```

Comment:
no need for a custom action flag - if you were separating the tabs into two distinct categories that would be one thing (like 'api key' and 'no api key'), but when a tab needs a special action taken, just check for `name == "VISUALIZE"` in the if statement. Right now this "custom action" is inextensible to future tabs other than visualize because each new tab would need something different done in its "special action" and you would end up checking its name anyway.